db/db_impl_compaction_flush.cc

```diff
@@ -1074,9 +1025,16 @@ Status DBImpl::RunManualCompaction(ColumnFamilyData* cfd, int input_level,
 
 Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
                              const FlushOptions& flush_options,
-                             FlushReason flush_reason, bool writes_stopped) {
+                             FlushReason flush_reason, bool writes_stopped,
+                             bool allow_write_stall) {
   Status s;
   uint64_t flush_memtable_id = 0;
+  if (!allow_write_stall) {
+    s = WaitUntilFlushWouldNotStallWrites(cfd);
+    if (!s.ok()) {
```

Comment:
What if the return value is `ok`, but the memtables have been flushed already? See my comments below.